Added support for $ifNull multi field support, $isNumber, $isArray

David-Taub opened the pull request:
Added support for $ifnull with multiple input fields, added in MongoDB v4.4
solves issue #748

pcorpet commented on tests/test__collection_api.py:
```diff
@@ -3787,6 +3787,20 @@ def test__aggregate_project_if_null(self):
+                ('b_and_c', {'$ifNull': ['$elem_b', '$elem_c', '<missing_both>']}),
+            ])}
+        ])
+        self.assertEqual([{'a_and_b': '<present_a>', 'b_and_a': '<present_a>',
```
Please break after `self.assertEqual(`

David-Taub replied:
Note that I added a helper `get_mongodb_version` that checks the version of MongoDB
I added the newer $ifNull syntax that is supported only on versions higher than v4.4 which is currently used in this project. 
When this project upgrades into MongoDB v5, the test will pass and the syntax can be added

Do you think the PR can be approved if the aggregation fails depending on the MongoDB version

pcorpet reviewed (changes requested):
Thanks for the feature.

Please add a test in test__mongomock to make sure the feature stays aligned with pymongo + MongoDB behavior.

David-Taub committed:
Add support for $ifnull with multiple input expressions (added in MongoDB v5).

David-Taub commented:
@pcorpet, Note that I added a helper get_mongodb_version that checks the version of MongoDB
I added the newer $ifNull syntax that is supported only on versions higher than v4.4 which is currently used in this project.
When this project upgrades into MongoDB v5, the test will pass and the syntax will be used and not fail (added a check for MongoDB version for that)

pcorpet commented on tests/test__mongomock.py:
```diff
@@ -3000,6 +3000,86 @@ def test__aggregate_count(self):
+            }
+        ])
+
+        @skipIf(
```
Please deindent otherwise this function is nested in the other one.

pcorpet replied:
Done.

pcorpet commented on mongomock/aggregate.py:
```diff
@@ -156,6 +156,10 @@
     '$arrayToObject',
     '$objectToArray',
 ]
+type_operators = [
```
Please add those in a separate PR.

pcorpet replied:
Done in #789

pcorpet reviewed (changes requested):
Thanks for your tests that revealed that we need more work.

I think that you're right that we could use a mechanism to handle different server versions in here, but I'd rather not introduce code that is not tested in the CI.

I think I'm going to first switch to MongoDB v5 for testing so that you can rebase and use that for the tests. Please remove the new helper and the skipIf parts.

pcorpet commented:
> Added support for $ifnull with multiple input fields, added in MongoDB v4.4 solves issue #748
Note that as opposed to what you wrote in your first comment here, the feature is only available *after* v4.4 not in 4.4.

pcorpet commented:
OK, this is done now, you have a `mongomock.SERVER_VERSION` that you can use
